pkg/controller/application/application_controller.go

```diff
@@ -396,19 +394,30 @@ func (c *Controller) processApplication(app *shipper.Application) error {
 		}
 	}
 
-	// existing rollout blocks in the system
-	var nsRolloutBlocks, globalRolloutBlocks []*shipper.RolloutBlock
-	if nsRolloutBlocks, err = c.rbLister.RolloutBlocks(app.Namespace).List(labels.Everything()); err != nil {
-		klog.Warningf("failed to list Namespace RolloutBlocks %s", err.Error())
+	rolloutBlocked, events, err := rolloutblock.BlocksRollout(c.rbLister, app)
+	for _, ev := range events {
+		c.recorder.Event(app, ev.Type, ev.Reason, ev.Message)
```

Comment:
Do you think it might make sense to pass the instance of recorder inside `rolloutblock.BlocksRollout`?